### PR TITLE
Sort output of agent-info

### PR DESCRIPTION
**Pull request by zytek**
Sorting output of agent-info makes it much more user-friendly.

Note: I'm not a Go developer so feel free to refactor it or leave a comment how to improve it. This is just a proposition.

### CODE REVIEW

File: command/agent_info.go [hunk 1]
```diff
@@ -53,14 +54,26 @@ func (c *AgentInfoCommand) Run(args []string) int {
 		return 1
 	}
 
+	// Sort and output agent info
 	var stats map[string]interface{}
 	stats, _ = info["stats"]
+	stats_keys := make([]string, 0, len(stats))
```
Comment (cbednarski) [inline command/agent_info.go]: You can simplify this as `stats_keys := make([]string, len(stats))`

File: command/agent_info.go [hunk 2]
```diff
@@ -53,14 +54,26 @@ func (c *AgentInfoCommand) Run(args []string) int {
...
+	for key := range stats {
+		stats_keys = append(stats_keys, key)
+	}
+	sort.Strings(stats_keys)
+
+	for _, key := range stats_keys {
+		c.Ui.Output(key)
+		d, _ := stats[key].(map[string]interface{})
+		d_keys := make([]string, 0, len(d))
```
Comment (cbednarski) [inline command/agent_info.go]: Likewise `d_keys := make([]string, len(d))`

File: command/agent_info.go [hunk 3]
```diff
@@ -53,14 +54,26 @@ func (c *AgentInfoCommand) Run(args []string) int {
...
 	stats, _ = info["stats"]
+	stats_keys := make([]string, 0, len(stats))
+	for key := range stats {
+		stats_keys = append(stats_keys, key)
+	}
+	sort.Strings(stats_keys)
+
+	for _, key := range stats_keys {
+		c.Ui.Output(key)
+		d, _ := stats[key].(map[string]interface{})
```
Comment (cbednarski) [inline command/agent_info.go]: Is there a better variable name for `d` here? It used to be short for `data` but this was replaced. Between the inner loop and the addition of `_keys` I'm having a hard time following which variable holds what data. I'm curious if we can name these a bit better like "stat_name" or "agent_info" or something.

File: command/agent_info.go (hunk 1 above)
Comment (c4milo) [inline command/agent_info.go]: Keep in mind that stats_keys is not a idiomatic variable notation in Go. It should be statsKeys (camelcase). My 2 cents.

**Comment by zytek** [comment]
Thank you for the comments, I've cleaned up variable names.

One thing got me curious, changing this:

```
-       statsKeys := make([]string, 0, len(stats))
+       statsKeys := make([]string, len(stats))
-               statsDataKeys := make([]string, 0, len(statsData))
+               statsDataKeys := make([]string, len(statsData))
```

Made agent-info return a bunch of empy = nil outputs as per this example 

```
serf
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
   = <nil>
  encrypted = false
  event_queue = 0
  event_time = 1
  failed = 0
  intent_queue = 0
  left = 0
  member_time = 1
  members = 1
  query_queue = 0
```

Can you explain that? Is it because im re-using variables or what?

**Comment by dadgar** [comment]
Closing in favor of #617.
@zytek you can see the fix in my commit. You were creating the slice at the desired size and then appending to it, instead of writing the keys in the created indexes.

**Comment by zytek** [comment]
Thx @dadgar  !

BTW, judging by https://github.com/hashicorp/consul/pull/969 - I guess both implementations are fine and it's just a matter of style ;-)

**Pull request closed without merging**